### PR TITLE
Migrate autocollection + standard metrics support from Azure monitor

### PR DESCRIPTION
**Pull request by lzchen**
From https://github.com/microsoft/opentelemetry-azure-monitor-python

### CODE REVIEW

**Comment by hectorhdzg** [comment]
@lzchen I thought we were going to create another project for this code because is not related to the actual exporter, did something change?

File: shared_requirements.txt [hunk 1]
```diff
@@ -156,6 +156,7 @@ opencensus-ext-threading
 opencensus-ext-azure>=0.3.1
 opentelemetry-api==0.16b1
 opentelemetry-sdk==0.16b1
+psutil>=5.8.0
```
Comment (rakshith91) [inline shared_requirements.txt:159]: do we need this in the shared_requirements?

Comment (lzchen) [reply shared_requirements.txt:159]: CI complains if not added.

**Comment by lzchen** [comment]
@hectorhdzg 
Another package will be created for the "azure monitor" offering to customers, but all the code will live in the azure-sdk repo. For this PR, we are just moving all the code over first until we decide on package specifics (name, location, etc).

**Review by hectorhdzg — APPROVED** [review]
LGTM

**Pull request merged**